src/c2pa/c2pa.py

```diff
@@ -675,7 +699,7 @@ def sign_file(
         # Ensure resources are cleaned up
         if 'builder' in locals():
             builder.close()
-        if 'signer' in locals():
+        if 'signer' in locals() and own_signer:
```

Comment:
Only clean up Signer if we created it for this function call only (otherwise the SIgner is owned by the calling code)